qa/rpc-tests/p2p-segwit.py

```diff
@@ -4,15 +4,93 @@
 # file COPYING or http://www.opensource.org/licenses/mit-license.php.
 """Test segwit transactions and blocks on P2P network."""
 
-from test_framework.mininode import *
-from test_framework.test_framework import BitcoinTestFramework
-from test_framework.util import *
-from test_framework.script import *
-from test_framework.blocktools import create_block, create_coinbase, add_witness_commitment, WITNESS_COMMITMENT_HEADER
-from test_framework.key import CECKey, CPubKey
-import time
-import random
 from binascii import hexlify
+import random
+import struct
+import time
+
+from test_framework.blocktools import (create_block,
+                                       create_coinbase,
+                                       add_witness_commitment,
+                                       WITNESS_COMMITMENT_HEADER)
+from test_framework.key import (CECKey,
+                                CPubKey)
+from test_framework.mininode import (CBlock,
+                                     CBlockHeader,
+                                     CInv,
+                                     COutPoint,
+                                     CTransaction,
+                                     CTxIn,
+                                     CTxInWitness,
+                                     CTxOut,
+                                     CTxWitness,
+                                     MAX_BLOCK_BASE_SIZE,
+                                     MSG_WITNESS_FLAG,
+                                     NODE_NETWORK,
+                                     NODE_WITNESS,
+                                     NetworkThread,
+                                     NodeConn,
+                                     NodeConnCB,
+                                     bytes_to_hex_str,
+                                     hash256,
+                                     hex_str_to_bytes,
+                                     mininode_lock,
+                                     msg_block,
+                                     msg_getdata,
+                                     msg_headers,
+                                     msg_inv,
+                                     msg_ping,
+                                     msg_pong,
+                                     msg_tx,
+                                     msg_witness_block,
+                                     msg_witness_tx,
+                                     ser_uint256,
+                                     ser_vector,
+                                     sha256,
+                                     uint256_from_str)
```

Comment:
One item per line is nice for diff-ing, though. Collapsing them can look somewhat nicer, however imagine having to insert an item in the middle and re-layouting. This will change *all* the lines, causing collisions with other patches.

For this reason we also use one-filename-per-line in makefiles, even though it results in long files.